tests/WorldStorage/FastNBT/Parse.cpp

```diff
@@ -0,0 +1,39 @@
+#include "Globals.h"
+#include "WorldStorage/FastNBT.h"
+
+int main() {
+	{
+		cParsedNBT nbt{nullptr, 0};
+
+		testassert(!nbt.IsValid());
+	}
+
+	{
+		cParsedNBT nbt{nullptr, 2};
+
+		testassert(!nbt.IsValid());
+	}
+	{
+		cParsedNBT nbt{"", 3};
+
+		testassert(!nbt.IsValid());
+	}
+	{
+		cParsedNBT nbt{nullptr, 3};
+
+		testassert(!nbt.IsValid());
+	}
+
+	{
+		char data[] = {10, 0, 0, 0x7F, -0x7F};
```

Comment:
I think `const void *` is better for the interface, so that it can parse anything, and internally cast to whatever datatype is best.